.github/workflows/lint-test.yml

```diff
@@ -33,144 +38,228 @@ jobs:
     # https://pdm-project.org/en/latest/usage/lockfile/
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-
-    # Instead of setup-env, we call the cache-pdm action here directly.
-    # This avoids having to install PDM, only to find out the cache is already up to date sometimes.
-    - name: Configure cache
-      uses: ./.github/actions/cache-pdm
-      with:
-        env-already-initialized: false
-
+      - uses: actions/checkout@v4
+
+      - name: Set up uv to lock dependencies
+        id: setup-python
+        uses: astral-sh/setup-uv@v6
+        with:
+          version: ${{ env.USE_UV_VERSION }}
+          enable-cache: true
+
+      - name: Restore uv lock
+        uses: actions/cache/restore@v4
+        id: cache-uv-lock
+        with:
+          key: uv-lock-resolved-${{ env.UV_RESOLUTION }}
+          restore-keys: |
+            uv-lock-resolved-${{ env.UV_RESOLUTION }}
+          path: uv.lock
+          enableCrossOsArchive: true
+
+      - name: Check if uv.lock changed
+        id: uv-lock-changed
+        continue-on-error: true
+        if: steps.cache-uv-lock.outputs.cache-hit == 'true' || steps.cache-uv-lock.outputs.cache-hit == 'false'
```

Comment:
This condition shouldn't be necessary, `lock --check` will also fail if the lockfile is missing entirely. Removing it would then also simplify the condition of the next step